### ISSUE TITLE
async Client with sync LocalCluster

### ISSUE THREAD

**Issue by crusaderky** [dask/distributed#3789]
When a Client is passed a LocalCluster and the asynchronous=True flag, it silently ignores the explicit flag and gets it from the cluster instead.

```python
>>> cluster = await distributed.LocalCluster(asynchronous=True)
>>> client = await distributed.Client(cluster, asynchronous=True)
>>> client.asynchronous
True
>>> cluster = distributed.LocalCluster()
>>> client = await distributed.Client(cluster, asynchronous=True)
>>> client.asynchronous
False
>>> cluster = distributed.LocalCluster()
>>> client = await distributed.Client(cluster.scheduler.address, asynchronous=True)
>>> client.asynchronous
True
```
This behaviour in turn is caused by these lines, which cause Client.loop to be forced to match Client.cluster.loop (even when it's been explicitly stated otherwise):
https://github.com/dask/distributed/blob/9f91653398aee681ba0010b1053f3bd1b415894b/distributed/client.py#L647-L653

I see in the code how a few things will break if the client and client.cluster end up having two different event loops. Not sure what the correct solution would be?

**Comment by quasiben** [comment]
The cluster and client loops should match and they do, I think.  Maybe we should instead, check in the `def asynchronous` property if `client.cluster` is set an then make a decision based on that rather than IOLoop.current()

**Comment by mrocklin** [comment]
I agree that silently ignoring arguments is probably bad.  I can't
currently come up with a use case of having two different event loops.  Do
you have a use case to share here?

I don't have any suggestions on how to resolve this currently.

**Comment by crusaderky** [comment]
I don't have a use case for two event loops, but when prototyping on jupyter notebook, if I need particular worker settings, I typically find it useful to write cluster and worker on two separate lines.
If the client is asynchronous, i frequently forget to set the asynchronous flag on the cluster.

Should I simply implement a check that, if address is a cluster, then the loop and asynchronous parameters must either be omitted or match those of the cluster?

**Comment by mrocklin** [comment]
Yes, I think that erring loudly in this case, perhaps with the workaround
that you suggested with the address would be a good solution.

**Comment by lmeyerov** [comment]
FWIW, we're seeing the reverse need in GPU land: sync client to an async cluster.

-- Most of our code is async client <> async cluster
-- Many libs (ex: blazingsql) assume sync client, which seems to require a sync cluster
-- We don't want 2 clusters, as dask_cudf / RMM takes ~300MB baseline per instance
-- ... so we'd like to occasionally generate a sync client to an existing async cluster
